Packs/VMware/Integrations/VMware/VMware.py

```diff
@@ -0,0 +1,412 @@
+                            raise task.info.error
+            version = update.version
+    finally:
+        if pcfilter:
+            pcfilter.Destroy()
+
+
+def soft_reboot(uuid):
+    vm = get_vm(uuid)
+    vm.RebootGuest()
+    return 'A request to reboot the guest has been sent.'
+
+
+def create_snapshot(args):
+    uuid = args['vm-uuid']
+    vm = get_vm(uuid)
+    d = str(datetime.now())
+    if args['memory'] == 'True':
+        mem = True
+    else:
+        mem = False
+    if args['quiesce'] == 'True':
+        qui = True
+    else:
+        qui = False
+    name = args.get('name', uuid + ' snapshot ' + d)
+    desc = args.get('description', 'Snapshot of VM UUID ' + uuid + ' taken on ' + d)
+    WaitForTask(vm.CreateSnapshot(name=name, description=desc, memory=mem, quiesce=qui))
+    return 'Snapshot ' + name + ' completed.'
+
+
+def revert_snapshot(name, uuid):
+    vm = get_vm(uuid)
+    snapObj = get_snapshots(vm.snapshot.rootSnapshotList, name)
+    if len(snapObj) == 1:
+        snapObj = snapObj[0].snapshot
+        WaitForTask(snapObj.RevertToSnapshot_Task())
+        ec = {
+            'VMWare(val.UUID && val.UUID === obj.UUID)': {
+                'UUID': uuid,
+                'Snapshot': 'Reverted to ' + name
+            }
+        }
+        return {
+            'ContentsFormat': formats['json'],
+            'Type': entryTypes['note'],
+            'Contents': ec,
+            'ReadableContentsFormat': formats['text'],
+            'HumanReadable': 'Reverted to snapshot ' + name + ' successfully.',
+            'EntryContext': ec
+        }
+    else:
+        return 'No snapshots found with name: ' + name + ' on VM: ' + uuid
+
+
+def get_snapshots(snapshots, snapname):
+    snapObj = []
+    for snapshot in snapshots:
+        if snapshot.name == snapname:
+            snapObj.append(snapshot)
+        else:
+            snapObj = snapObj + get_snapshots(snapshot.childSnapshotList, snapname)
+    return snapObj
+
+
+def get_events(uuid, event_type):
+    vm = get_vm(uuid)
+    hr = []
+    content = si.RetrieveServiceContent()
+    eventManager = content.eventManager
+    filter = vim.event.EventFilterSpec.ByEntity(entity=vm, recursion="self")
+    filterSpec = vim.event.EventFilterSpec()
+    ids = event_type.split(',')
+    filterSpec.eventTypeId = ids
+    filterSpec.entity = filter
+    eventRes = eventManager.QueryEvents(filterSpec)
+    for e in eventRes:
+        hr.append({
+            'Event': e.fullFormattedMessage,
+            'Created Time': e.createdTime.strftime("%Y-%m-%d %H:%M:%S")
+        })
+    return {
+        'ContentsFormat': formats['json'],
+        'Type': entryTypes['note'],
+        'Contents': hr,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown('VM ' + vm.summary.config.name + ' Events', hr) if hr else 'No result were found'
+    }
+
+
+def change_nic_state(args):
+    uuid = args['vm-uuid']
```

Comment:
this is existing pack and integration. The only added command is `change_nic_state`
coordinate with @GavrielFilippov